packages/runtime-test/src/nodeOps.ts

```diff
@@ -224,8 +224,28 @@ function nextSibling(node: TestNode): TestNode | null {
   return parent.children[i + 1] || null
 }
 
-function querySelector(): any {
-  throw new Error('querySelector not supported in test renderer.')
+function querySelector(selector: string): any {
+  if (selector[0] !== '#') {
+    throw new Error(
+      'querySelector only supports simple IDs in the test renderer.'
+    )
+  }
+
+  // FIXME: Ideally we'd use something like JSDOM, run querySelector, and map back to the test target element (… uhh somehow)
```

Comment:
I don't think it makes sense to support `querySelector` here because most of the time it's used are querying elements **not** created by Vue. So any use cases that relies on `querySelector` should only be tested in the tests for the runtime that supports it.

Reply:
Okay, that makes sense.

Thinking through some things:
- Would you prefer to test portals with integration tests and JSDOM APIs, then? (… perhaps in addition to these?)
- Or, is this id-only support suitable enough for test purposes.
- Or, is there another option that might be more suitable for testing portals?

Reply:
Tests for portals in `runtime-core` should handle direct HostNode as target only. Then there should be an integration test in `runtime-dom` where string selector target is tested.

Reply:
In the current implementation it doesn't look like you can specify a HostNode as a target, just a selector. I'll look into adding that functionality (…assuming I understand this correctly) and updating the tests.

See: https://github.com/vuejs/vue-next/blob/0fe88017825b04cee8131b983b24f0f0742cf052/packages/runtime-core/src/createRenderer.ts#L623